noir-projects/noir-contracts/contracts/token_contract/src/main.nr

```diff
@@ -691,46 +674,58 @@ contract Token {
         context.storage_write(slot + aztec::protocol_types::point::POINT_LENGTH as Field, setup_log);
     }
 
-    // TODO(#7728): even though the funded_amount should be a U128, we can't have that type in a contract interface due
+    // TODO(#7728): even though the max_fee should be a U128, we can't have that type in a contract interface due
     // to serialization issues.
     // docs:start:complete_refund
+    /// Executed as a public teardown function and is responsible for completing the refund in a private fee payment
+    /// flow.
     #[public]
     #[internal]
-    fn complete_refund(fee_payer_slot: Field, user_slot: Field, funded_amount: Field) {
+    fn complete_refund(fee_recipient: AztecAddress, user_slot: Field, max_fee: Field) {
         // TODO(#7728): Remove the next line
-        let funded_amount = U128::from_integer(funded_amount);
+        let max_fee = U128::from_integer(max_fee);
```

Comment:
In the FPC we're doing arithmetic on Fields, whereas here we're doing it on U128's. Which is preferred?